sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleIterator.scala

```diff
@@ -306,10 +316,12 @@ class RapidsShuffleIterator(
     }
 
     val blockedStart = System.currentTimeMillis()
-    val result = resolvedBatches.take()
+    var result: Option[ShuffleClientResult] = None
+
+    result = pollForResult(timeoutSeconds)
     val blockedTime = System.currentTimeMillis() - blockedStart
```

Comment:
Not directly related to this change, but why are we not unconditionally updating the time spent blocked here?  Seems wrong to measure time spent waiting yet not always report it in the metric.  Does not need to be fixed as part of this, can track in a followup issue if necessary.

Reply:
good point, will update `blockedTime`

Reply:
handled in https://github.com/NVIDIA/spark-rapids/pull/732/commits/57fd60f5c2d103429e3b89c58152636f4797e075